src/py/flwr/driver/app.py

```diff
@@ -197,9 +186,9 @@ def update_client_manager(
         for node_id in new_nodes:
             client_proxy = DriverClientProxy(
                 node_id=node_id,
-                driver=driver,
+                driver=cast(GrpcDriver, driver.grpc_driver),
                 anonymous=False,
-                workload_id=workload_id,
+                workload_id=cast(int, driver.workload_id),
```

Comment:
Shouldn't we introduce a getter method on `Driver` for this? This would prevent us from doing the `cast` above

Reply:
Make sense. I've updated the `DriverClientProxy` to use `Driver` instead of `GrpcDriver`.